source/agora/consensus/state/ValidatorSet.d

```diff
@@ -360,7 +360,37 @@ public class ValidatorSet
 
     /***************************************************************************
 
-        Gets the number of active validators at the block height.
+        Get the utxo_key used for enrollment for the given public key
+
+        Params:
+            height = block height to fetch utxo_key
+            public_key = public key of the validator
+
+        Returns:
+            Return UTXO key as Hash
+
+    ***************************************************************************/
+
+    public Hash getEnrollmentForKey (in Height height, in PublicKey public_key) @trusted nothrow
+    {
+        try
+        {
+            auto results = this.db.execute("SELECT key FROM validator " ~
+                "WHERE public_key = ? AND enrolled_height >= ? AND enrolled_height < ? ",
+                public_key, this.minEnrollmentHeight(height), height);
+            if (!results.empty && results.oneValue!(byte[]).length != 0)
+                return Hash(results.front.peek!(char[])(0));
+        }
+        catch (Exception ex)
+        {
+            log.error("ManagedDatabase operation error: {}", ex.msg);
+        }
+        return Hash.init;
```

Comment:
This is only needed because the Ledger iterates on `PublicKey` instead of `Hash` right ?

Reply:
Currently we use PublicKey for signatures map. This can be removed after the move to using utxo everywhere is completed.